smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy

```diff
@@ -307,71 +316,101 @@ def getEcobeeDevices() {
 	def thermostatList = [:]
 	def remoteSensors = [:]
 	def switchList = [:]
-	try {
-		// First get thermostats and thermostat remote sensors
-		def bodyParams = [
-				selection: [
-						selectionType:  "registered",
-						selectionMatch: "",
-						includeRuntime: true,
-						includeSensors: true
+	def isThermostatPolled = false
+	def isSwitchesPolled = false
+	def pollAttempt = 1
+	// try obtain devices twice, in case authToken needs to be refreshed
+	while (!(isThermostatPolled && isSwitchesPolled) && (pollAttempt < 3)) {
+		try {
+			// First get thermostats their remote sensors
+			if (!isThermostatPolled) {
```

Comment:
I do this in case this poll was ok but the switch poll fails in order to avoid polling thermostats a second time when we only need to poll the switches